include/d3dx12Residency.h

```diff
@@ -110,6 +110,65 @@ namespace D3DX12Residency
 		class ResidencyManagerInternal;
 	}
 
+	// Used to contain waits that must be satisfied before a pinned ManagedObject can be unpinned.
+	class PinWaits
+	{
+	public:
+		class PinWait
+		{
+		public:
+			PinWait(ID3D12Fence* pFence, UINT64 value)
+				: m_pFence(pFence), m_value(value)
+			{
+				m_pFence->AddRef();
+			}
+
+			PinWait(const PinWait& pinWait)
+			{
+				m_pFence = pinWait.m_pFence;
+				m_pFence->AddRef();
+				m_value = pinWait.m_value;
+			}
+
+			~PinWait()
+			{
+				m_pFence->Release();
+			}
+
+			ID3D12Fence* m_pFence;
+			UINT64 m_value;
+		};
+
+		~PinWaits() { ClearPinWaits(); }
+
+		void ClearPinWaits()
+		{
+			m_pinWaits.clear();
+		}
+
+		void AddPinWaits(UINT NumSync, _In_reads_(NumSync) UINT64* pSignalValues, _In_reads_(NumSync) ID3D12Fence** ppFences)
+		{
+			m_pinWaits.reserve(m_pinWaits.size() + NumSync); // throw( bad_alloc);
+			for (UINT i(0); i < NumSync; ++i)
+			{
+				m_pinWaits.push_back(PinWait( ppFences[i], pSignalValues[i] ));
+			}
+		}
+
+		bool CheckPinWaits()
+		{
+			m_pinWaits.erase(std::remove_if(m_pinWaits.begin(), m_pinWaits.end(), [](auto& pinWait)

```

Comment:
This form of `erase` only removes one element. You need to erase from `remove_if` to `end()`.

Reply:
Thank you for catching this one.